src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -1,38 +1,130 @@
+            "snapshot/own",
+            timeout = 45.seconds, //todo change these?
+            tag = "snapshot/obj/snapshot"
+          )(C)
+          resp.map((id, _))
+      }
+    } yield chunkedPeerProposalMaps
+
+  def updatePeerProps(fetchedProposals: Seq[(Id, RecentSnapshot)]) =
+    peersProposals.modify { m =>
+      val updatedProposals = fetchedProposals.foldLeft(m) {
+        case (prevPeerProps, (id, recentSnap)) =>
+          val proposalsAtHeight: Seq[(Id, Seq[RecentSnapshot])] = prevPeerProps.getOrElse(recentSnap.height, Seq())
+          val hasMadeProposal = proposalsAtHeight.exists(_._1 == id) //todo Observation of new proposal
+          if (!hasMadeProposal) prevPeerProps.updated(recentSnap.height, proposalsAtHeight :+ (id, Seq(recentSnap)))
+          else prevPeerProps
+      }
+      (updatedProposals, updatedProposals)
+    }
 
-  // TODO: Check for the majority snapshot and store under a variable
-  def recalculateMajoritySnapshot(): F[Unit] = ???
+  def fetchAndSetPeersProposals() = {
+    val fetchedProposals = getProposals().map { proposals =>
+      proposals.map(RedownloadService.deSerProps).flatMap {
+        case (id, recentSnaps) => recentSnaps.map(snap => (id, snap))
+      }
+    }
+    fetchedProposals.flatMap(updatePeerProps)
+  }
 
-  // TODO: Check for the alignment ownSnapshots <-> majoritySnapshot and call redownload if needed
-  def checkForAlignmentWithMajoritySnapshot(): F[Unit] = ???
+  def recalculateMajoritySnapshot(): F[(Seq[RecentSnapshot], Set[Id])] =
+    for {
+      peerProps <- peersProposals.get
+      ownProps <- ownSnapshots.get
+      ownPropsNormalized = ownProps.mapValues(snap => List((dao.id, Seq(snap))))
+      allProposals = peerProps.mapValues(_.toList) |+| ownPropsNormalized
+      allProposalsNormalized = allProposals.values.flatten.toList
+      peers <- LiftIO[F].liftIO(dao.readyPeers) //todo need testing around join leave, this could cause divergence
+      majority = MajorityStateChooser.chooseMajorWinner(peers.keys.toSeq, allProposalsNormalized)
+      groupedProposals = allProposalsNormalized
+        .groupBy(_._1)
+        .map { case (id, tupList) => (id, tupList.flatMap(_._2)) }
+        .toList
+      snapsThroughMaj = majority.map(maj => MajorityStateChooser.getAllSnapsUntilMaj(maj._2, groupedProposals))
+      majNodeIds = majority.flatMap(maj => MajorityStateChooser.chooseMajNodeIds(maj._2, allProposalsNormalized))
+    } yield (snapsThroughMaj.map(_.sortBy(-_.height)).getOrElse(Seq()), majNodeIds.map(_.toSet).getOrElse(Set()))
+
+  def checkForAlignmentWithMajoritySnapshot(): F[Option[List[RecentSnapshot]]] =
+    for {
+      peers <- LiftIO[F].liftIO(dao.readyPeers)
+      majSnapsIds <- recalculateMajoritySnapshot()
+      ownSnaps <- ownSnapshots.get
+      diff = HealthChecker.compareSnapshotState(majSnapsIds, ownSnaps.values.toList)
+      shouldRedownload = HealthChecker.shouldReDownload(ownSnaps.values.toList, diff)
```

Comment:
Here as well. `should redownload` should be an internal thing of `redownload service`